### DISCUSSION TITLE
Complicated moves in chess does not work

### DISCUSSION THREAD

**Discussion by salastro** [lowlighter/metrics#1335] [Q&A]
* [Castling move](https://en.wikipedia.org/wiki/Castling) shows only the king moving without the rook taking the correct position, it shows the correct position after moving the rook again (it suddenly jumps over the king).
* [Promotion moves](https://en.wikipedia.org/wiki/Promotion_(chess)) does not show the new piece, only the pawn changes to, for example, queen when it is moved again.

**Comment by lowlighter** [comment]
Hi 👋 !

Thanks for reporting this, will be tracked in #1340
